common/djangoapps/xmodule_django/models.py

```diff
@@ -104,6 +107,16 @@ def to_python(self, value):
             return None
 
         if isinstance(value, basestring):
+            if value.endswith('\n'):
+                # An opaque key with a trailing newline has leaked into the DB.
+                # Log and strip the value.
+                log.warning('{}:{}:{}:to_python: Invalid key: {}. Removing trailing newline.'.format(
```

Comment:
not that I can imagine this being an issue, but should this be a unicode string?